scripts/set_assignees.py

```diff
@@ -181,14 +181,21 @@ def process_pr(gh, maintainer_file, number):
             existing_reviewers |= set(r.get_page(page))
             page += 1
 
-        for c in collab:
+        # check for reviewers that remove themselves from list of reviewer and
+        # do not attempt to add them again based on MAINTAINER file.
+        self_removal = []
+        for event in pr.get_issue_events():
+            if event.event == 'review_request_removed':
+                self_removal.append(event.actor)
```

Comment:
I don't think it's enough, as someone may remove someone else's request for review, and I don't think that's the same as someone removing *themselves*.  I think you need to check that event.actor  == requested_reviewer.login

Reply:
> I don't think it's sufficient of a check: someone may remove someone else's request for review, 

Technically correct but... a little bit paranoid maybe?

Who is lucky enough to have someone else help with their spam? :-)

Reply:
I think it's not uncommon for someone to inadvertently push the wrong change set to a PR, have the bot pull in dozens of irrelevant reviewers, and then manually clean up the mess.

Reply:
This was also the agreement on the process WG meeting. Only self-removal should result in not being re-added.